expression/builtin_compare_test.go

```diff
@@ -279,23 +279,27 @@ func (s *testEvaluatorSuite) TestGreatestLeastFuncs(c *C) {
 		},
 		{
 			[]interface{}{"123a", "b", "c", 12},
-			float64(123), float64(0), false, false,
+			"c", "12", false, false,
 		},
 		{
 			[]interface{}{tm, "123"},
 			curTimeString, "123", false, false,
 		},
 		{
 			[]interface{}{tm, 123},
-			curTimeInt, int64(123), false, false,
+			curTimeString, "123", false, false,
+		},
+		{
+			[]interface{}{tm, tmWithFsp},
+			curTimeWithFspString, curTimeString, false, false,
 		},
 		{
-			[]interface{}{tm, "invalid_time_1", "invalid_time_2", tmWithFsp},
-			curTimeWithFspString, "invalid_time_1", false, false,
+			[]interface{}{tm, "invalid_time_a", "invalid_time_b", tmWithFsp},
+			curTimeWithFspString, curTimeString, false, false,
 		},
 		{
-			[]interface{}{tm, "invalid_time_2", "invalid_time_1", tmWithFsp},
-			curTimeWithFspString, "invalid_time_2", false, false,
+			[]interface{}{"invalid_time_a", "invalid_time_b"},
+			"invalid_time_b", "invalid_time_a", false, false,
```

Comment:
this test case does not make any sense?

Reply:
Why not include your test cases in [this issue](https://github.com/pingcap/tidb/issues/17994)?